### PR TITLE
[#98091344] Ansible role for deploying Telegraf

### PR DESCRIPTION
**Pull request by jimconner**
**What**

We wish to deploy the [Telegraf](https://github.com/influxdb/telegraf) stats collection agent for influxdb using an ansible role

This ansible role will:
- Fetch the amd64 version of the .deb package from the influxdb.org download site
- Install the package using dpkg
- Ensure the service is enabled and started
- Generate a config file for telegraf from a template

**How to review this PR**

I've created the PR to be reviewed in the following context:
- I need the telegraf stats collection client for influxdb to be deployed and configured by ansible

**Who should review this PR**
- Anyone in the core team except for @actionjack because he worked on this with me.

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+# tasks file for ansible-playbook-telegraf
+#
+
+- name: fetch telegraf installation package
+  get_url: url=http://get.influxdb.org/telegraf/telegraf_{{ telegraf_version }}_amd64.deb dest=/tmp/telegraf_{{ telegraf_version }}_amd64.deb mode=0440
+
+- name: install telegraf package
+  command: dpkg --skip-same-version -i /tmp/telegraf_{{ telegraf_version }}_amd64.deb
```
Comment (dcarley) [inline tasks/main.yml]: It looks like the [apt module](http://docs.ansible.com/apt_module.html#options) takes a `deb` option which is probably safer than shelling out.

File: tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  register: dpkg_result
+  changed_when: "dpkg_result.stdout.startswith('Selecting')"
+
+- name: start telegraf service and ensure that it is enabled
+  service: name=telegraf enabled=yes state=started
```
Comment (dcarley) [inline tasks/main.yml:12]: Would it be better if this came after the config task so that we don't start it twice (once with the default config)?

Comment (government-paas-user) [inline tasks/main.yml:12]: We did start out with this config the other way around, but it complains that the destination directory didn't exist - that directory is created by telegraf when it first starts up. We could have it before, but we'd have to create the directory with ansible if we want to swap the order over.

File: tasks/main.yml (hunk 1 above)
Comment (jimconner) [inline tasks/main.yml]: New commit https://github.com/alphagov/ansible-playbook-telegraf/commit/cc6ee5205d614af5ddf1e1c73bbe517d45a09d8b resolves this issue.

**Comment by keymon** [comment]
:+1: Looks good for me, as it installs the software and it is running.

I wonder if we can get instructions of how get something running and integrated for testing, like how to install a test environment with influxdb and a graphana to display some metrics, and configure it with telegraf. I guess probably that will be part of the next PR.

**Comment by keymon** [comment]
I will merge.

**Pull request merged**